components/ConnectionSelection.go

```diff
@@ -140,50 +143,69 @@ func NewConnectionSelection(connectionForm *ConnectionForm, connectionPages *mod
 	return cs
 }
 
-func (cs *ConnectionSelection) Connect(connection models.Connection) {
+func (cs *ConnectionSelection) Connect(connection models.Connection) *tview.Application {
 	if MainPages.HasPage(connection.URL) {
 		MainPages.SwitchToPage(connection.URL)
-		App.Draw()
-	} else {
-		cs.StatusText.SetText("Connecting...").SetTextColor(app.Styles.TertiaryTextColor)
+		return App.Draw()
+	}
+
+	ctx, cancelFn := context.WithCancel(context.Background())
+
+	if len(connection.Commands) > 0 {
+		cs.StatusText.SetText("Running commands...").SetTextColor(app.Styles.TertiaryTextColor)
 		App.Draw()
 
-		var newDbDriver drivers.Driver
+		// This is super arbitrary, but allows you to use '${port}' in your commands
+		// which will be replaced with a random port number between 5000 and 6000 to
+		// avoid conflicts when connecting to multiple databases from the same machine.
+		port := 5000 + rand.Intn(1000)
+		connection.URL = strings.ReplaceAll(connection.URL, "${port}", fmt.Sprintf("%d", port))
```

Comment:
This is a bit dirty and not fail proof

Try something like this

https://gist.github.com/sevkin/96bdae9274465b2d09191384f86ef39d

Thanks @sekvin BTW

Reply:
Guess this indeed is a little less dirty 😉 Added an updated version of the example and used that to acquire a port.